compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -867,6 +879,21 @@ private void recursivelyDefineVariables(BLangVariable variable, SymbolEnv blockE
         }
     }
 
+    private void recursivelySetFinalFlag(BLangVariable variable) {
+        switch (variable.getKind()) {
```

Comment:
Don't we need to check for `ERROR_VARIABLE` here? Is it final by default?

I think it is better to add a default clause as well which throws an error to make sure that no unexpected value gets ignored here. WDYT?

Reply:
@Shan1024 I thought i come under generic variable, as we are dealing with variables when restructured. Let me add a test case for that and verify

Reply:
you are right. it fails for error variable. Fixed it and added a test